tools/juliapkg/src/result.jl

```diff
@@ -704,6 +705,7 @@ function execute(stmt::Stmt, params::DBInterface.StatementParams = ())
     if Threads.nthreads() != 1
         # When we have additional worker threads, don't execute using the main thread
         while duckdb_execution_is_finished(stmt.con.handle) == false
+            Base.yield()
```

Comment:
At line 700 you're using `Threads.nthreads()` (sorry, cannot comment on that line since it's not changed).
But since Julia 1.9 or even some patches of Julia 1.8, this function is deprecated, since there are 2 threadpools: default and interactive. You don't want to schedule execution on interactive threads, as that might jeopardize interactivity of the process (e.g. HTTP serving).
I think you want to use `Threads.nthreads(:default)`.

Reply:
Also, if deadlock is fixed, perhaps now we can utilize all threads for execution, and not skip the first one?

Reply:
These 1.8 patches are not public so you can only experiment with the interactive thread pool using the public 1.9+ version

Reply:
Please disregard my comment about nthreads, I just checked and it is actually backward compatible, so it will return the same as with :default.